controllers/cart.controller.js

```diff
@@ -0,0 +1,78 @@
+const { Cart } = require('../models/cart.model');
+
+const getAllProductsInCart = async (req, res) => {
+  try {
+    const { userID } = req.user;
+
+    const cart = await Cart.findById(userID).populate('products._id');
+    const cartProducts = cart.products.map((productItem) => {
+      return { ...productItem._id._doc, qty: productItem.qty };
+    });
+
+    const finalCart = { _id: cart._id, products: cartProducts };
+
+    res.json({ success: true, cart: finalCart });
+  } catch (error) {
+    res.json({ success: false, message: 'Error retrieving cart', errorMessage: error.message });
+  }
+};
+
+const addProductInCart = async (req, res) => {
+  try {
+    const { userID } = req.user;
+    let { product } = req.body;
+
+    const user = await Cart.findById(userID);
+    if (!user) {
+      const newCart = new Cart({
+        _id: userID,
+        products: [{ _id: product._id, qty: 1 }],
+      });
+      await newCart.save();
+      res.json({ success: true, message: 'New cart created and product added', newCart });
+    } else {
+      const newProduct = { _id: product._id, qty: 1 };
+      user.products.push(newProduct);
+      await user.save();
+      res.json({ success: true, message: 'Product added to exisitng cart', user });
+    }
+  } catch (error) {
+    res.json({ success: false, message: 'Error adding product to cart', errorMessage: error.message });
+  }
+};
+
+const updateProductInCart = async (req, res) => {
+  try {
+    let { qty } = req.body;
+    const { userID } = req.user;
+    const { productID } = req.params;
+
+    const user = await Cart.findById(userID);
```

Comment:
hey pranshu, on line 7 you wrote `const cart = await Cart.findById(userID).populate('products._id');` and on line 50 and line 69, you're doing `const user = await Cart.findById(userID);`.

Both these lines are essentially doing the same thing which is finding by userId in cart schema but you named two different variables i.e `cart` and `user` which can slightly make readability complicated. If you want to, you can give a consistent name to the variable in this file.